pkg/agent/billing/billing.go

```diff
@@ -96,22 +97,33 @@ func (mc *MetricsCollector) Run(
 	logger *zap.Logger,
 	store VMStoreForNode,
 ) error {
-	logger = logger.Named("collect")
-
 	collectTicker := time.NewTicker(time.Second * time.Duration(mc.conf.CollectEverySeconds))
 	defer collectTicker.Stop()
 	// Offset by half a second, so it's a bit more deterministic.
 	time.Sleep(500 * time.Millisecond)
 	accumulateTicker := time.NewTicker(time.Second * time.Duration(mc.conf.AccumulateEverySeconds))
 	defer accumulateTicker.Stop()
 
+	// Create a new taskgroup to manage mc.sink.Run() -- we want to run the event senders in the
```

Comment:
I would go one step further here:
 - Extract a new function `.runCollector()`
 - In this function do 
```
tg := taskgroup.NewGroup(logger)

tg.Go("collector-run", func(logger *zap.Logger) error {
	err := mc.collectorRun(ctx)
	cancelSink()
	return err
})

tg.Go("sink-run", func(logger *zap.Logger) error {
	mc.sink.Run(tg.Ctx())
	return nil
})

return tg.Wait()
```

See also https://github.com/neondatabase/autoscaling/pull/1235.

Reply:
Makes sense, implemented in f0ba926ca848f93dffb9f6916ea7de137c9e3959